flink-python/src/main/java/org/apache/flink/table/runtime/operators/python/aggregate/arrow/batch/BatchArrowPythonGroupWindowAggregateFunctionOperator.java

```diff
@@ -0,0 +1,216 @@
+import org.apache.flink.table.data.TimestampData;
+import org.apache.flink.table.data.binary.BinaryRowData;
+import org.apache.flink.table.functions.AggregateFunction;
+import org.apache.flink.table.functions.python.PythonFunctionInfo;
+import org.apache.flink.table.runtime.operators.window.TimeWindow;
+import org.apache.flink.table.runtime.operators.window.grouping.HeapWindowsGrouping;
+import org.apache.flink.table.runtime.operators.window.grouping.WindowsGrouping;
+import org.apache.flink.table.runtime.util.RowIterator;
+import org.apache.flink.table.types.logical.RowType;
+
+import java.util.LinkedList;
+
+/**
+ * The Batch Arrow Python {@link AggregateFunction} Operator for Group Window Aggregation.
+ */
+@Internal
+public class BatchArrowPythonGroupWindowAggregateFunctionOperator
+	extends AbstractBatchArrowPythonAggregateFunctionOperator {
+
+	private static final long serialVersionUID = 1L;
+
+	/**
+	 * The Infos of the Window.
+	 * 0 -> start of the Window.
+	 * 1 -> end of the Window.
+	 * 2 -> row time of the Window.
+	 */
+	private final int[] namedProperties;
+
+	/**
+	 * The row time index of the input data.
+	 */
+	private final int inputTimeFieldIndex;
+
+	/**
+	 * The window elements buffer size limit used in group window agg operator.
+	 */
+	private final int maxLimitSize;
+
+	/**
+	 * The window size of the window.
+	 */
+	private final long windowSize;
+
+	/**
+	 * The sliding size of the sliding window.
+	 */
+	private final long slideSize;
+
+	private transient WindowsGrouping windowsGrouping;
+
+	/**
+	 * The GenericRowData reused holding the property of the window, such as window start, window
+	 * end and window time.
+	 */
+	private transient GenericRowData windowProperty;
+
+	/**
+	 * The JoinedRowData reused holding the window agg execution result.
+	 */
+	private transient JoinedRowData windowAggResult;
+
+	/**
+	 * The queue holding the input groupSet with the TimeWindow for which the execution results
+	 * have not been received.
+	 */
+	private transient LinkedList<Tuple2<RowData, TimeWindow>> inputKeyAndWindow;
+
+	public BatchArrowPythonGroupWindowAggregateFunctionOperator(
+		Configuration config,
+		PythonFunctionInfo[] pandasAggFunctions,
+		RowType inputType,
+		RowType outputType,
+		int inputTimeFieldIndex,
+		int maxLimitSize,
+		long windowSize,
+		long slideSize,
+		int[] namedProperties,
+		int[] groupKey,
+		int[] groupingSet,
+		int[] udafInputOffsets) {
+		super(config, pandasAggFunctions, inputType, outputType, groupKey, groupingSet, udafInputOffsets);
+		this.namedProperties = namedProperties;
+		this.inputTimeFieldIndex = inputTimeFieldIndex;
+		this.maxLimitSize = maxLimitSize;
+		this.windowSize = windowSize;
+		this.slideSize = slideSize;
+	}
+
+	@Override
+	public void open() throws Exception {
+		userDefinedFunctionOutputType = new RowType(
+			outputType.getFields().subList(groupingSet.length, outputType.getFieldCount() - namedProperties.length));
+		super.open();
```

Comment:
Putting super.open() at the end of this method? Although it doesn't affect the correctness, it makes the code more readable.